src/Raven.Server/Documents/Handlers/BulkInsertHandler.cs

```diff
@@ -206,6 +207,53 @@ public override int Execute(DocumentsOperationContext context)
                 }
                 return NumberOfCommands;
             }
+
+            public DynamicJsonValue Serialize()
+            {
+                var dJsonCommands = new DynamicJsonValue[NumberOfCommands];
+                for (var i = 0; i < NumberOfCommands; i++)
+                {
+                    dJsonCommands[i] = BatchRequestParser.CommandToDynamicJson(Commands[i], Database);
+                }
+
+                var ret = new DynamicJsonValue
+                {
+                    [nameof(Commands)] = dJsonCommands
+                };
+
+                return ret;
+            }
+
+            public static MergedInsertBulkCommand Deserialize(BlittableJsonReaderObject mergedCmdReader, DocumentDatabase database, JsonOperationContext context,
+                Logger logger)
+            {
+                if (mergedCmdReader.TryGet(nameof(Commands), out BlittableJsonReaderArray commandsReader) == false)
+                {
+                    throw new Exception($"Cant read {nameof(Commands)} from {nameof(MergedInsertBulkCommand)}");
```

Comment:
Do not throw `Exception` directly. Use `InvalidOperationException` or some such

Reply:
Cant vs Can't

Reply:
done